src/vanilla/Model/CompositeTypeTS.cs

```diff
@@ -33,6 +33,35 @@ public override Property Add(Property item)
             return result;
         }
 
+        private List<CompositeTypeTS> polymorphicUnionCases;
+
+        /// <summary>Contains the immediate polymorphic children of this class.</summary>
+        public List<CompositeTypeTS> PolymorphicUnionCases
+        {
+            get
+            {
+                if (polymorphicUnionCases == null)
+                {
+                    if (BaseIsPolymorphic)
+                    {
+                        polymorphicUnionCases =
+                            CodeModel.ModelTypes
+                                .Where(m => m.BaseModelType == this)
+                                .Cast<CompositeTypeTS>()
+                                .ToList();
```

Comment:
So this only looks for types that derive from this type, right? This won't include this type?

Reply:
Yeah, this is so you can `type FishUnion = Fish | SalmonUnion | SharkUnion`-- by treating "this type" as a special case, you can make sure you reference the interface instead of the tagged union.

Reply:
Think this would be better named... `ImmediatePolymorphicSubtypes` or something..